tests/kafkatest/tests/streams/streams_cooperative_rebalance_upgrade_test.py

```diff
@@ -0,0 +1,191 @@
+                rebalance_mode_msg = self.cooperative_enabled_msg
+
+            self.set_props(processor, upgrade_version)
+            node = processor.node
+            with node.account.monitor_log(processor.STDOUT_FILE) as stdout_monitor:
+                with node.account.monitor_log(processor.LOG_FILE) as log_monitor:
+                    processor.start()
+                    # verify correct rebalance mode either turned off for upgrade or enabled after upgrade
+                    log_monitor.wait_until(rebalance_mode_msg,
+                                           timeout_sec=60,
+                                           err_msg="Never saw '%s' message " % rebalance_mode_msg + str(processor.node.account))
+
+                # verify rebalanced into a running state
+                rebalance_msg = current_phase + self.running_state_msg
+                stdout_monitor.wait_until(rebalance_msg,
+                                          timeout_sec=60,
+                                          err_msg="Never saw '%s' message " % rebalance_msg + str(
+                                              processor.node.account))
+
+                # verify processing
+                verify_processing_msg = current_phase + self.processing_message
+                stdout_monitor.wait_until(verify_processing_msg,
+                                          timeout_sec=60,
+                                          err_msg="Never saw '%s' message " % verify_processing_msg + str(
+                                              processor.node.account))
+
+    def verify_processing(self, processor, pattern):
+        self.logger.info("Verifying %s processing pattern in STDOUT_FILE" % pattern)
+        with processor.node.account.monitor_log(processor.STDOUT_FILE) as monitor:
+            monitor.wait_until(pattern,
+                               timeout_sec=60,
+                               err_msg="Never saw processing of %s " % pattern + str(processor.node.account))
+
+    def get_tasks_for_processor(self, processor):
+        retries = 0
+        while retries < 5:
+            found_tasks = list(processor.node.account.ssh_capture("grep TASK-ASSIGNMENTS %s | tail -n 1" % processor.STDOUT_FILE, allow_fail=True))
+            self.logger.info("Returned %s from assigned task check" % found_tasks)
+            if len(found_tasks) > 0:
+                task_string = str(found_tasks[0]).strip()
+                self.logger.info("Converted %s from assigned task check" % task_string)
+                processor.set_tasks(task_string)
+                return
+            retries += 1
+            time.sleep(1)
+        return
+
+    def set_props(self, processor, upgrade_from=None):
+        processor.SOURCE_TOPIC = self.source_topic
+        processor.SINK_TOPIC = self.sink_topic
+        processor.TASK_DELIMITER = self.task_delimiter
+        processor.REPORT_INTERVAL = self.report_interval
+        processor.UPGRADE_FROM = upgrade_from
```

Comment:
Since we now let the log4j to print the upgrade_from string already, do we still need this and to print within the state listener any more?

Reply:
Not sure I follow here.  We need the `upgrade_from` field for setting in the  `CooperativeRebalanceUpgradeService` service to set the streams config.

Reply:
My bad, I thought the `upgradePhase-Starting a REBALANCE` message is used somewhere in verification and thought with the augmented log4j it can now be replaced, but now I see it is only for debugging purposes.